### PR TITLE
Fix problem when use default port hidden

### PR DESCRIPTION
**Pull request by fbrissi**
When you use the hidden default port (80 or 443), that split gets the host name and assigns it to the port variable this value.

Using location.port to get correct port.

### CODE REVIEW

**Comment by fbrissi** [comment]
When using https, the default port is not 80, I added a configuration in the utility to use the correct default port.

I need this because my applications failed to connect the socket.

**Comment by fbrissi** [comment]
Do you have any predictions for this merge.
If you need any adjustments you could tell me, this is important where I go.

**Comment by barakluminati** [comment]
Hi,

I am confused. Which merge are you referring to?

Nolan

**Comment by fbrissi** [comment]
I opened this merge because I'm using the luminati proxy manager on port 443 and on the console I get several errors, because it tries to access port 80.

File: src/pub/whitelist_ips.js [hunk 1]
```diff
@@ -33,9 +34,9 @@ export default class Whitelist_ips extends Pure_component {
 }
 
 const Admin_steps = ({ip})=>{
-    const port = location.port||80;
```
Comment (fbrissi) [inline src/pub/whitelist_ips.js:36]: @barakluminati here, when no has port, and the protocol is ssl, the deafault port is 443 and no 80

File: src/pub/ws.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 'use strict'; /*jslint browser:true, react:true, es6:true*/
 import {EventTarget} from 'event-target-shim';
+import { default_ports } from './util';
 import setdb from '../../util/setdb.js';
 
 class Ws_wrapper extends EventTarget {
     constructor(){
         super();
         this.url = location.hostname;
-        this.port = location.port||80;
```
Comment (fbrissi) [inline src/pub/ws.js:10]: @barakluminati here, when no has port, and the protocol is ssl, the deafault port is 443 and no 80

**Comment by barakluminati** [comment]
Hi,

By default all http requests are sent thru port 80. As well, all https requests
are sent thru port 443. If you would like to send https thru port 80, please
contact your account manager individually as this will require a port
opening/compliance approval.

Nolan

**Comment by fbrissi** [comment]
No, I do not want.
But the main branch does not consider that port 443 can be used.
That's why I did this fix.

See, when, I login, the redirect send to port 80. But should be for 443

image

**Comment by fbrissi** [comment]
You didn't understand my fix?
Could you please point out the problem in the code I wrote?

**Comment by fbrissi** [comment]
If you think something is wrong I can fix it.

**Comment by fbrissi** [comment]
It should also be here at port 443

image

File: src/pub/whitelist_ips.js [hunk 3]
```diff
@@ -33,9 +34,9 @@ export default class Whitelist_ips extends Pure_component {
...
+    const port = location.port||default_ports[location.protocol];
```
Comment (fbrissi) [inline src/pub/whitelist_ips.js:37]: This fix checks the protocol to be sent to the correct port

File: src/pub/ws.js [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-        this.protocol = location.protocol=='https:' ? 'wss' : 'ws';
+        this.port = location.port||default_ports[location.protocol];
```
Comment (fbrissi) [inline src/pub/ws.js:11]: This fix checks the protocol to be sent to the correct port

**Comment by fbrissi** [comment]
@romankoblov 
@giladno 
@lee-elenbaas 

If you can, I would like more opinions on this.
This issue is very important for my implementation.
Thank you.

**Comment by barakluminati** [comment]
I understand. I'm happy to help you but I am confused as to what you need.
That's why I asked you to refer your tech questions to our tech support team:

lumsupport@luminati.io

Thx,
Nolan

**Comment by fbrissi** [comment]
Solved in https://github.com/luminati-io/luminati-proxy/commit/39396b83ddab222691ed958cd787b655e4000e88

**Comment by fbrissi** [comment]
Solved in https://github.com/luminati-io/luminati-proxy/commit/39396b83ddab222691ed958cd787b655e4000e88

**Pull request closed without merging**